src/components/widgets/stocks/stocks.jsx

```diff
@@ -0,0 +1,102 @@
+import useSWR from "swr";
+import { useState } from "react";
+import { useTranslation } from "next-i18next";
+import { FaChartLine } from "react-icons/fa6";
+import { BiError } from "react-icons/bi";
+
+import Error from "../widget/error";
+import Container from "../widget/container";
+import PrimaryText from "../widget/primary_text";
+import WidgetIcon from "../widget/widget_icon";
+import Raw from "../widget/raw";
+
+function Widget({ options }) {
+  const { t, i18n } = useTranslation();
+
+  const [viewingPercentChange, setViewingPercentChange] = useState(false);
+
+  const { color } = options;
+
+  const { data, error } = useSWR(
+    `/api/widgets/stocks?${new URLSearchParams({ lang: i18n.language, ...options }).toString()}`,
+  );
+
+  if (data?.error) {
+    return (
+      <Container options={options} additionalClassNames="information-widget-error">
+        <PrimaryText>{t(data?.error)}</PrimaryText>
+        <WidgetIcon icon={BiError} size="l" />
+      </Container>
+    );
+  }
+
+  if (error) {
+    return <Error options={error} />;
+  }
```

Comment:
```suggestion
  if (data?.error || error) {
    return (
      <Container options={options} additionalClassNames="information-widget-error">
        <PrimaryText>{error ? error : data?.error}</PrimaryText>
        <WidgetIcon icon={BiError} size="l" />
      </Container>
    );
  }
```

Combine these two, also cant translate the error

Reply:
Should I simplify to use the Error component? 

```
  if (error || data?.error) {
    return <Error options={options} />;
  }

```

https://github.com/gethomepage/homepage/blob/810c321881d9acdde3666c4cac0fd6700d12735b/src/components/widgets/openmeteo/openmeteo.jsx#L20 

https://github.com/gethomepage/homepage/blob/810c321881d9acdde3666c4cac0fd6700d12735b/src/components/widgets/longhorn/longhorn.jsx#L15

https://github.com/gethomepage/homepage/blob/810c321881d9acdde3666c4cac0fd6700d12735b/src/components/widgets/glances/glances.jsx#L33

Reply:
Yea sounds good